src/modules/ekf2/EKF2.cpp

```diff
@@ -349,17 +349,40 @@ void EKF2::Run()
 
 				if (_ekf.setEkfGlobalOrigin(latitude, longitude, altitude)) {
 					// Validate the ekf origin status.
-					uint64_t origin_time {};
-					_ekf.getEkfGlobalOrigin(origin_time, latitude, longitude, altitude);
-					PX4_INFO("%d - New NED origin (LLA): %3.10f, %3.10f, %4.3f\n",
-						 _instance, latitude, longitude, static_cast<double>(altitude));
+					PX4_INFO("%d - New NED origin (LLA): %3.10f, %3.10f, %4.3f", _instance,
+						 _ekf.global_origin().getProjectionReferenceLat(),
+						 _ekf.global_origin().getProjectionReferenceLon(),
+						 static_cast<double>(_ekf.getEkfGlobalOriginAltitude()));
 
 				} else {
-					PX4_ERR("%d - Failed to set new NED origin (LLA): %3.10f, %3.10f, %4.3f\n",
+					PX4_ERR("%d - Failed to set new NED origin (LLA): %3.10f, %3.10f, %4.3f",
 						_instance, latitude, longitude, static_cast<double>(altitude));
 				}
 			}
 		}
+
+	} else if (_multi_mode) {
+		estimator_selector_status_s ekf_selection;
+		vehicle_local_position_s lpos;
+
+		if (_estimator_selector_status_sub.update(&ekf_selection)
+		    && ekf_selection.primary_instance != _instance
+		    && _vehicle_local_position_sub.updated()
+		    && _vehicle_local_position_sub.update(&lpos)
+		    && lpos.ref_timestamp > 0
+		    && (abs(lpos.ref_lat - _ekf.global_origin().getProjectionReferenceLat()) > 1e-9
```

Comment:
When the global position isn't valid, the `ref_xx` is set to NAN; the `abs(lpos.ref_lat - _ekf.global_origin().getProjectionReferenceLat()) > 1e-9` should return `abs(NAN) > 1e-9` = `false` as needed, but I think it would be safer to check for the NAN explicitly